src/recipient_dashboard/tests.py

```diff
@@ -480,3 +466,82 @@ def test_search_no_results(self):
 
         # Check that no items are returned
         self.assertEqual(len(response.context["donations"]), 0)
+
+
+# add test_search_address, test_search_address_radius, test_search_address_no_results
+def test_search_address(self):
+    """Test searching donations by address."""
+    params = {"address": "123 Test St"}
+    url = reverse("recipient_dashboard") + "?" + urlencode(params)
+    response = self.client.get(url)
+    donation_items = [donation.food_item for donation in response.context["donations"]]
+
+    # Check that the correct number of items are returned
+    self.assertEqual(len(response.context["donations"]), 2)
+
+    # Check that the correct items are returned
+    self.assertIn("Biryani", donation_items)
+    self.assertIn("Shawarma", donation_items)
+    self.assertNotIn("Pasta", donation_items)
+
+
+def test_search_address_radius(self):
+    """Test searching donations by address and radius."""
+    params = {"address": "123 Test St", "radius": "5"}
+    url = reverse("recipient_dashboard") + "?" + urlencode(params)
+    response = self.client.get(url)
+    donation_items = [donation.food_item for donation in response.context["donations"]]
+
+    # Check that the correct number of items are returned
+    self.assertEqual(len(response.context["donations"]), 2)
+
+    # Check that the correct items are returned
+    self.assertIn("Biryani", donation_items)
+    self.assertIn("Shawarma", donation_items)
+    self.assertNotIn("Pasta", donation_items)
+
+
+def test_search_address_no_results(self):
+    """Test searching donations by address with no matches."""
+    params = {"address": "456 Nonexistent St"}
+    url = reverse("recipient_dashboard") + "?" + urlencode(params)
+    response = self.client.get(url)
+
+    # Check that no items are returned
+    self.assertEqual(len(response.context["donations"]), 0)
+
+
+# In a new test class for utils.py
+
+
+class UtilsTests(TestCase):
```

Comment:
Any specific reason for adding this class?

Reply:
Removed. I had it as a place holder to test urls.py